lib/phoenix/router/helpers.ex

```diff
@@ -223,7 +241,11 @@ defmodule Phoenix.Router.Helpers do
       end
 
       def unquote(:"#{helper}_path")(conn_or_endpoint, unquote(opts), unquote_splicing(vars), params) do
-        path(conn_or_endpoint, segments(unquote(segs), params, unquote(bins)))
+        path(conn_or_endpoint, segments!(unquote(segs), params, unquote(bins), [
+          opts: unquote(opts),
+          helper: unquote(helper),
+          var_length: length(unquote(vars))
+        ]))
```

Comment:
Sorry for nitpicking but I would rather just pass them all as arguments. Or a tuple of three elements. Build a list of things that are always present is probably the slowest way to go about this. :)